app/src/main/java/com/github/featuredetectandroid/ui/AppLayout.kt

```diff
@@ -0,0 +1,80 @@
+package com.github.featuredetectandroid.ui
+
+import android.content.Context
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.PointMode
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.platform.LocalDensity
+import com.github.featuredetectandroid.utils.KeypointDetectionAlgorithm
+import com.github.featuredetectandroid.utils.PreferencesManager
+
+@Composable
+fun AppLayout(
+    context: Context,
+    outputViewModel: OutputViewModel,
+    preferencesManager: PreferencesManager,
+    selectedAlgorithm: String,
+    onAlgorithmSelected: (String) -> Unit
```

Comment:
You see, the majority of this arguments is only used inside `Menu`'s `onSelected`. Let's just pass what's you're passing there instead of `onAlgorithmSelected` (you can leave this naming, though) and get rid of the most of the arguments.

Reply:
After this, I think, we can also replace passing `ViewModel` here with three arguments we use from it. This way we can remove `isCameraPermissionGranted` property from `ViewModel` as it'll be only used in `MainActivity`.

Reply:
fixed